src/app/clusters/bindings/BindingManager.cpp

```diff
@@ -171,23 +176,26 @@ CHIP_ERROR BindingManager::NotifyBoundClusterChanged(EndpointId endpoint, Cluste
         EmberBindingTableEntry entry;
 
         if (emberGetBinding(i, &entry) == EMBER_SUCCESS && entry.type != EMBER_UNUSED_BINDING && entry.local == endpoint &&
-            entry.clusterId == cluster)
+            (entry.clusterId.HasValue() || entry.clusterId.Value() == cluster))
```

Comment:
This line does not make sense: if !HasValue() then we will crash.  Presumably this is meant to be:
```suggestion
            (!entry.clusterId.HasValue() || entry.clusterId.Value() == cluster))
```
and we should add a test (followup fine) for the case when cluster has no value.